src/types/datetime.rs

```diff
@@ -160,13 +162,13 @@ pub trait PyTimeAccess {
 }
 
 /// Trait for accessing the components of a struct containing a tzinfo.
-pub trait PyTzInfoAccess {
+pub trait PyTzInfoAccess<'py> {
     /// Returns the tzinfo (which may be None).
     ///
     /// Implementations should conform to the upstream documentation:
     /// <https://docs.python.org/3/c-api/datetime.html#c.PyDateTime_DATE_GET_TZINFO>
     /// <https://docs.python.org/3/c-api/datetime.html#c.PyDateTime_TIME_GET_TZINFO>
-    fn get_tzinfo(&self) -> Option<&PyTzInfo>;
+    fn get_tzinfo(&self) -> Option<Bound<'py, PyTzInfo>>;
 }
```

Comment:
Following the rebase onto the now-public `Bound` API I've decided to just make (small) breaking changes to the `PyTzInfoAccess` trait, particularly in changing its return type.

Reply:
Naturally I forgot to update our own code for this change...

Reply:
Maybe the docs should explain how to get from the new return type to a GIL ref? Or will this happen elsewhere and we can just explain it once in the migration guide?

Reply:
I would prefer not add it to this API doc, as I think we'll repeat it in several places. Instead yes I'll start working on the documentation which will explain how to migrate.